interfaces/dbus/backend.go

```diff
@@ -179,6 +288,54 @@ func addContent(securityTag string, snippet string, content map[string]osutil.Fi
 	}
 }
 
+// snapNameFromServiceFile returns the snap name for the D-Bus service activation file.
+func snapNameFromServiceFile(filename string) (owner string, err error) {
+	content, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return "", err
+	}
+	snapKey := []byte("\nX-Snap=")
+	pos := bytes.Index(content, snapKey)
+	if pos == -1 {
+		return "", nil
+	}
+	snapName := content[pos+len(snapKey):]
+	pos = bytes.IndexRune(snapName, '\n')
+	if pos != -1 {
+		snapName = snapName[:pos]
+	}
+	return string(snapName), nil
+}
+
+func (b *Backend) deriveContentServices(snapInfo *snap.Info, servicesDir string, services map[string]*Service) (globs []string, content map[string]osutil.FileState, err error) {
+	snapName := snapInfo.InstanceName()
+	// Prime globs with the already existing service activation
+	// files associated with this snap.
+	globs, err = snapServiceActivationFiles(servicesDir, snapName)
+	if err != nil {
+		return nil, nil, err
+	}
+	content = make(map[string]osutil.FileState)
+	for _, service := range services {
+		filename := service.BusName + ".service"
+		if old, err := snapNameFromServiceFile(filepath.Join(servicesDir, filename)); err != nil {
```

Comment:
Again voicing my concern for using non-state files as as input to the interface logic.